auth-service/src/test/java/com/petclinic/auth/User/AuthServiceUserServiceTests.java

```diff
@@ -146,4 +156,39 @@ void encrypt_password_before_persistence() {
         assertFalse(saved.getPassword().isEmpty());
         assertNotEquals(saved.getPassword(), userIDLessDTO.getPassword());
     }
+
+    @Test
+    @DisplayName("When creating user, send verification email")
+    void send_email_on_register() {
+
+        AtomicInteger callCount = new AtomicInteger();
+        AtomicReference<Mail> mailRef = new AtomicReference<>();
+
+        when(mailService.sendMail(any())).then(args -> {
+            callCount.incrementAndGet();
+            final Mail mail = args.getArgument(0, Mail.class);
+            mailRef.set(mail);
+            return "Email sent to " + mail.getMessage();
+        });
+
+        final UserIDLessDTO userIDLessDTO = new UserIDLessDTO(USER, PASS, EMAIL);
+        final User saved = userService.createUser(userIDLessDTO);
+
+        assertEquals(1, callCount.get());
+        assertTrue(mailRef.get().getMessage().contains("Your verification link: "));
+    }
+
+    @Test
+    @DisplayName("Given user, generate verification email")
+    void generate_verification_email() {
+
+        final UserIDLessDTO userIDLessDTO = new UserIDLessDTO(USER, PASS, EMAIL);
+        final User saved = userService.createUser(userIDLessDTO);
+
+        final Mail mail = userService.generateVerificationMail(saved);
+
+        System.out.println(mail);
```

Comment:
unnecessary print to console?

Reply:
Ah shit I saw this too late.
I'll make a note and remove it in my next PR